src/request.rs

```diff
@@ -682,14 +680,14 @@ impl Request {
     /// ```
     /// use http_types::{Method, Request, Url};
     ///
-    /// let mut req = Request::get("https://example.com");
+    /// let mut req = Request::get("https://example.com").unwrap();
     /// req.set_body("Hello, Nori!");
     /// assert_eq!(req.method(), Method::Get);
     /// ```
-    pub fn get<U>(url: U) -> Self
+    pub fn get<U>(url: U) -> Result<Self, Error>
     where
         U: TryInto<Url>,
-        U::Error: std::fmt::Debug,
+        U::Error: std::fmt::Debug + Into<anyhow::Error>,
```

Comment:
All of these calls should instead use `http_types::Error`. Or even better: `http_types::Result`:

```suggestion
    pub fn get<U>(url: U) -> crate::Result<Self>
    where
        U: TryInto<Url>,
        U::Error: std::fmt::Debug,